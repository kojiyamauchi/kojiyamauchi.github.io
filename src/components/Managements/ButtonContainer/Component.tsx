/*
  Button Container Component.
*/

import { shallowEqual, useSelector } from 'react-redux'

import { CloseButton } from '@/components/Presentations/CloseButton'
import { DynamicComponentOpenButton } from '@/components/Presentations/OpenButton'
import { store } from '@/ducks'
import { modalToggle } from '@/ducks/Modal'
import { StoreTypes } from '@/types/type'

import styles from './style.module.scss'

export const modalOpen = (): void => void store.dispatch(modalToggle({ isOpen: true }))
export const modalClose = (): void => void store.dispatch(modalToggle({ isOpen: false }))

type Props = {
  storybook?: boolean
}

export const ButtonContainer: React.VFC<Props> = ({ storybook = false }): JSX.Element => {
  const isOpen = useSelector<StoreTypes, boolean>((state) => state.modalStore.isOpen, shallowEqual)
  return (
    <div className={`${styles.wrapper} button-container`}>
      <DynamicComponentOpenButton modalState={isOpen} modalOpen={modalOpen} storybook={storybook} />
      <CloseButton modalState={isOpen} modalClose={modalClose} />
    </div>
  )
}
