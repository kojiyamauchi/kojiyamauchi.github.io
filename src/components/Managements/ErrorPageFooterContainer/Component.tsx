/*
  Error Page Footer Container Component.
*/

import { Copyright } from '@/components/Presentations/Copyright'
import styles from './style.module.scss'

export const ErrorPageFooterContainer: React.VFC = (): JSX.Element => {
  return (
    <footer className={`${styles.wrapper} error-page-footer-container`}>
      <Copyright parent="error" />
    </footer>
  )
}
