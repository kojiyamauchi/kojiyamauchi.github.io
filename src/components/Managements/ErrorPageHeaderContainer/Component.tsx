/*
  Error Page Header Container Component.
*/

import { ErrorPageHeading } from '@/components/Presentations/ErrorPageHeading'
import { ErrorPageNavigation } from '@/components/Presentations/ErrorPageNavigation'
import styles from './style.module.scss'

export const ErrorPageHeaderContainer: React.VFC = (): JSX.Element => {
  return (
    <header className={`${styles.wrapper} error-page-header-container`}>
      <ErrorPageHeading />
      <ErrorPageNavigation />
    </header>
  )
}
