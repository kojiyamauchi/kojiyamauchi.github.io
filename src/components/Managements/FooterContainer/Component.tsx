/*
  Footer Container Component.
*/

import { Copyright } from '@/components/Presentations/Copyright'

import styles from './style.module.scss'

export const FooterContainer: React.VFC = (): JSX.Element => {
  return (
    <footer className={`${styles.wrapper} footer-container`}>
      <Copyright parent="footer" />
    </footer>
  )
}
