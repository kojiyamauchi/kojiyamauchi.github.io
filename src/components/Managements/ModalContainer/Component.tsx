/*
  Modal Container Component.
*/

import { ResizeObserver } from '@juggle/resize-observer'
import { Canvas } from '@react-three/fiber'
import { Suspense } from 'react'

import { Copyright } from '@/components/Presentations/Copyright'
import { JumpToOldPortfolio } from '@/components/Presentations/JumpToOldPortfolio'
import { DynamicComponentLinkCube } from '@/components/Presentations/LinkCube'
import { ThanksForVisiting } from '@/components/Presentations/ThanksForVisiting'

import { useModal } from './Hooks'
import styles from './style.module.scss'

export const ModalContainer: React.VFC = (): JSX.Element => {
  const { myComponent, cubesProps, generatePositionValue, openAboutMe, copyrightHover } = useModal()

  return (
    <div className={`${styles.wrapper} modal-container`} ref={myComponent}>
      <div className={`${styles['graffiti-wrapper']} graffiti-wrapper`}>
        <ThanksForVisiting copyrightHover={copyrightHover} openAboutMe={openAboutMe} />
        <JumpToOldPortfolio copyrightHover={copyrightHover} openAboutMe={openAboutMe} />
      </div>
      <div className={`${styles['canvas-wrapper']} canvas-wrapper`}>
        <Canvas resize={{ polyfill: ResizeObserver }}>
          <Suspense fallback={null}>
            <directionalLight position={[1, 1, 1]} />
            <ambientLight intensity={0.65} />
            {cubesProps.map((cubeProps, index) => (
              <DynamicComponentLinkCube
                key={index}
                textureName={cubeProps.textureName}
                linkUrl={cubeProps.linkUrl}
                positionValue={generatePositionValue[index]}
                rotate={cubeProps.rotate}
              />
            ))}
          </Suspense>
        </Canvas>
      </div>
      <div className={`${styles['copyright-wrapper']} copyright-wrapper`}>
        <Copyright className="copyright" parent="modal" />
      </div>
    </div>
  )
}
