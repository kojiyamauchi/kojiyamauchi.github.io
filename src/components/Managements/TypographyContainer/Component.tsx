/*
  Typography Container Component.
*/

import { Typography } from '@/components/Presentations/Typography'
import { SubLetter } from '@/components/Presentations/SubLetter'
import { useTypography } from './Hooks'
import styles from './style.module.scss'

type Props = {
  visited: boolean
}

export const TypographyContainer: React.VFC<Props> = ({ visited }): JSX.Element => {
  const { myComponent } = useTypography(visited)

  return (
    <div className={`${styles.wrapper} typography-container`} ref={myComponent}>
      <Typography visited={visited} />
      <SubLetter visited={visited} />
    </div>
  )
}
