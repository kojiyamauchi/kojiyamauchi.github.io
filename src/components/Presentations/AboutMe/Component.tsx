/*
  About Me Component.
*/

import styles from './style.module.scss'

type Props = {
  closeAboutMe: () => void
}

export const AboutMe: React.FC<Props> = ({ closeAboutMe }): JSX.Element => {
  return (
    <div className={`${styles.wrapper} about-me`}>
      <h1 className={`${styles['primary-heading']} primary-heading`}>
        <span className="is-en">Koji Yamauchi</span>&nbsp;&nbsp;|&nbsp;&nbsp;<span className="is-jp">ヤマウチ コウジ</span>
      </h1>
      <p className={`${styles.letter} letter is-en`}>
        I am mainly involved in web front-end development and production from Based in Tokyo, Japan.
        <br />
        (Some back-end development and design work)
        <br />
        In the front-end area, we can handle everything from styling, such as UI implementation and design implementation, to interaction implementation,
        animation implementation, and API communication, so please feel free to contact us.
        <br />
        My skills, TypeScript / JavaScript / React / Next.js / Node.js / Webpack / Gulp / GitHub / Sass / CSS / HTML / VSCode / Figma(a little) / Sketch(a
        little) / Photoshop / Illustrator.
        <br />
        Currently, past projects is private.
        <br />
        If want you see please following contact us.
        <br />
        I’ll get back to you.
        <br />
        Loves &amp; fun making things.<span className="is-emoji">✌️</span>
      </p>
      <p className={`${styles.letter} letter is-jp`}>
        東京都内にて、主にWebフロントエンドの開発・制作に携わっております。
        <br />
        (Node.js開発・デザイン制作も少々)
        <br />
        フロントエンド領域においては、UI実装・デザイン実装等のスタイリング周りから、インタラクション実装・アニメーション実装・API通信まで承る事が可能ですので、お気軽にご連絡下さい。
        <br />
        使用言語・開発環境等は、TypeScript・JavaScript・React・Next.js・Node.js・Webpack・Gulp・GitHub・Sass・CSS・HTML・VSCode・Figma(少々)・Sketch(少々)・Photoshop・Illustratorになります。
        <br />
        現在、制作実績は非公開になりますので、下記よりお問い合わせ下さい。
        <br />
        折り返しご連絡させていただきます。
      </p>
      <div className={`${styles['mail-wrapper']} mail-wrapper`}>
        <a className={`${styles['button-mail']} button-mail`} href="mailto:all.the.small.things@me.com">
          <svg className={`${styles['icon-arrow-mail']} icon-arrow-mail`} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M4.01 6.03l7.51 3.22-7.52-1 .01-2.22m7.5 8.72L4 17.97v-2.22l7.51-1M2.01 3L2 10l15 2-15 2 .01 7L23 12 2.01 3z" />
          </svg>
          CONTACT US
        </a>
      </div>
      <div className={`${styles['back-wrapper']} back-wrapper`}>
        <button className={`${styles['button-back']} button-back`} onClick={closeAboutMe}>
          <svg className={`${styles['icon-arrow-back']} icon-arrow-back`} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
          </svg>
          HIDE ABOUT ME
        </button>
      </div>
    </div>
  )
}
