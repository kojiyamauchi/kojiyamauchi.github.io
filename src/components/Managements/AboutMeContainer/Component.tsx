/*
  About Me Container Component.
*/

import { useRouter } from 'next/router'
import { shallowEqual, useSelector } from 'react-redux'

import { AboutMe } from '@/components/Presentations/AboutMe'
import { Copyright } from '@/components/Presentations/Copyright'
import { store } from '@/ducks'
import { aboutMeToggle } from '@/ducks/AboutMe'
import { StoreTypes } from '@/types/type'

import styles from './style.module.scss'

export const closeAboutMe = (): void => void store.dispatch(aboutMeToggle({ isOpen: false }))

export const AboutMeContainer: React.VFC = (): JSX.Element => {
  const router = useRouter()
  const aboutMe = useSelector<StoreTypes, boolean>((state) => state.aboutMeStore.isOpen, shallowEqual)

  return (
    <div
      className={`${styles.wrapper} about-me-container ${aboutMe ? 'is-active' : ''}`}
      style={{ backgroundImage: `${router ? `url("${router.basePath}/webps/logo.webp")` : 'url("webps/logo.webp")'}` }}
    >
      <AboutMe closeAboutMe={closeAboutMe} />
      <div className={`${styles['copyright-wrapper']} copyright-wrapper`}>
        <Copyright className="copyright" parent="about-me" />
      </div>
    </div>
  )
}
