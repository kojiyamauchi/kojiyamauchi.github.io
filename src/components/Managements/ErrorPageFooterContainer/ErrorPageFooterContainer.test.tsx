/**
 * @jest-environment jsdom
 */
/*
  Error Page Footer Container Component Unit Test.
*/

import '@testing-library/jest-dom/extend-expect'

import { render /* ,screen */ } from '@testing-library/react'

import { ErrorPageFooterContainer } from './Component'
// import userEvent from '@testing-library/user-event'

describe('Error Page Footer Container Component Unit Test', () => {
  it('Snap Shot Testing', () => {
    const { asFragment } = render(<ErrorPageFooterContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
