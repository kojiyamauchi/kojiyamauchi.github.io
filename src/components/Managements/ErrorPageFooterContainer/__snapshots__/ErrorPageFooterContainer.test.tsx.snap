// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page Footer Container Component Unit Test Snap Shot Testing 1`] = `
<DocumentFragment>
  <footer
    class="wrapper error-page-footer-container"
  >
    <a
      class="wrapper undefined is-error"
      href="http://kojiyamauchi.com"
      rel="noreferrer"
      target="_blank"
    >
      Â© 2021 All The Small Things, All Rights Reserved.
    </a>
  </footer>
</DocumentFragment>
`;
